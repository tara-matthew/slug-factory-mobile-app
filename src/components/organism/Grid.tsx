import React, { useEffect } from "react";
import { FlatList, StyleSheet } from "react-native";
import Card from "../molecule/Card";

const Grid = (props) => {
    useEffect(() => {
        console.log(props.prints);
    }, []);
    const renderItem = ({ item }) => {
        return (
            // <View style={{marginRight: 10}}>
            <Card item={ item } />
            // </View>
        );
    };
    return (
        <FlatList
            className="mb-8"
            // style={styles.container}
            contentContainerStyle={ styles.grid }
            columnWrapperStyle={ { justifyContent: "space-between", gap: 20 } }
            data={ props.prints }
            numColumns={ 2 }
            initialNumToRender={ 5 }
            renderItem={ renderItem }
            keyExtractor={ item => item.id }
        />
    );
};

const styles = StyleSheet.create({
    grid: {
        padding: 20,
    },
    container: {
        // gap: 20
    },
    cardContainer: {
        // gap: 20,
        // flex: 1, // Allows the Card to take up available space
        // marginRight: 10, // Or you could use padding in the grid
        // marginBottom: 20,
    },
    card: {
        marginRight: 20,
    },
});

export default Grid;
