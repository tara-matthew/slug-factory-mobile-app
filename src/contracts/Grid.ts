import { IImage } from "./Image";

export interface IGridProps {
    items: IGridItem[];
}

export interface IGridItem {
    title: string;
    images: IImage[];
}
