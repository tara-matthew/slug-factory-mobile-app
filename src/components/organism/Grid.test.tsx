import React from "react";
import renderer from "react-test-renderer";
import Grid from "./Grid";
it("Renders correctly", () => {
    const prints = [
        {
            created_at: "2024-09-16T14:21:07.000000Z",
            description: "test",
            filament_brand: {

            },
            filament_colour: {

            },
            filament_material: {

            },
            id: 2000,
            images: [
                {
                    id: 1,
                    url: "test-url.com",
                },
            ],
            infill_percentage: null,
            title: "Gorgeous Concrete Shoes ",
            type: "PrintedDesign",
            user_id: 2001,
            is_favourite: false,
        },
    ];
    const tree = renderer.create(<Grid items={ prints } />).toJSON();
    expect(tree).toMatchSnapshot();
});

it.todo("Displays the correct number of items");
