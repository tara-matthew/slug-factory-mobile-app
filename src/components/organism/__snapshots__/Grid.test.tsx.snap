// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<RCTScrollView
  className="mb-8"
  contentContainerStyle={
    {
      "padding": 20,
    }
  }
  data={
    [
      {
        "created_at": "2024-09-16T14:21:07.000000Z",
        "description": "test",
        "filament_brand": {},
        "filament_colour": {},
        "filament_material": {},
        "id": 2000,
        "images": [
          {
            "id": 1,
            "url": "test-url.com",
          },
        ],
        "infill_percentage": null,
        "is_favourite": false,
        "title": "Gorgeous Concrete Shoes ",
        "type": "PrintedDesign",
        "user_id": 2001,
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  initialNumToRender={5}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            {
              "gap": 20,
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "#d0cadb",
                "flex": 1,
                "marginBottom": 20,
                "shadowRadius": 2,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              className="relative"
            >
              <Image
                placeholder={
                  {
                    "blurhash": undefined,
                  }
                }
                source={
                  {
                    "uri": "undefined/test-url.com",
                  }
                }
                style={
                  {
                    "height": 200,
                    "width": "100%",
                  }
                }
                transition={1000}
              />
              Icon: favorite-outline, Size: 32, Color: white
              <Text
                adjustsFontSizeToFit={true}
                numberOfLines={2}
                style={
                  {
                    "fontSize": 18,
                    "height": 50,
                    "padding": 8,
                  }
                }
              >
                Gorgeous Concrete Shoes 
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
